### PR TITLE
"device" parameter

### PR DESCRIPTION
**Pull request by d9394**
1, change temporary_directory to '/dev/shm' , I feel it is faster than '/tmp'
2, add a "device" parameter to the source config and source command, if "device": "" , it means not use "device" in the sdrcommand.

### CODE REVIEW

File: owrx/source.py [hunk 1]
```diff
@@ -378,15 +378,23 @@ def activateProfile(self, profile_id=None):
 
 class RtlSdrSource(SdrSource):
     def getCommand(self):
-        return "rtl_sdr -s {samp_rate} -f {center_freq} -p {ppm} -g {rf_gain} -"
+        command = "rtl_sdr -s {samp_rate} -f {center_freq} -p {ppm} -g {rf_gain}"
+        if self.rtlProps["device"] != "" :
```
Comment (D0han) [inline owrx/source.py]: Again, it makes "device" not optional.
```suggestion
        if "device" in self.rtlProps:
```

**Comment by jketterl** [comment]
Alright... I've been busy with work for the last week, sorry for the late reaction.

For the device changes: I agree with @D0han , checking for empty strings is kind of patchy. His suggested change is the python way of checking if a key is present in a dictionary, and it should work on a PropertyManager. I'd prefer this, but please let me know if there's any problems with it that I'm not seeing right now. We can fix the PropertyManager, if that is in the way.

For the temporary directory: I have read up on the FHS, and from what I can tell, it is not part of the standard, hence not guaranteed to be available. Also, from what I gathered, its intended use does not match 100% with what openwebrx would be using it for. This should not prevent you from using from it (after all, it's configurable for a reason), but I'd like to provide reasonable defaults here, and as far as I can tell, `/tmp` should have a greater chance of working "out of the box".

**Comment by D0han** [comment]
Instead of `/dev/shm` [FHS 3.0](https://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.pdf) describes `/run/` under 3.15 for such case. If you are not convinced to use it as a default, it would be nice to get a comment in config file near that default value with some short description of `/run` or pointer to such explanation somewhere.

**Comment by jketterl** [comment]
`/run` seems to be more in-line with the standard, but it will need a little work since (I suppose) it's good practice to open up a separate sub-directory there (which is probably a good idea, even when running on /tmp). I'd go with that.

**Comment by jketterl** [comment]
Reading the FHS, i stumbled across this: "/run should not be writable for unprivileged users" - I think that will be a problem. I don't think it's a good idea to run openwebrx as root, which would basically require an admin to set up write access beforehand. ugh... who came up with that...

**Pull request closed without merging**